src/WidgetBase.ts

```diff
@@ -271,17 +285,42 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			delete this._previousProperties[propertyName];
 		});
 
-		const diffPropertiesResult = this.diffProperties(this._previousProperties, properties);
-		this._properties = assign(diffPropertiesResult.properties, diffPropertyResults);
+		const diffPropertiesDecorators = this.getDecorator('diffProperties');
+		if (diffPropertiesDecorators.length) {
+			const diffDecorator = diffPropertiesDecorators[diffPropertiesDecorators.length - 1];
```

Comment:
Is this to deal with `@diffProperties` being added twice? That interesting re extended widgets that the behaviour for diffProperties added by the super class can be "overridden".